From 858ff4702685fd18a8b03373816de0c4236995f4 Mon Sep 17 00:00:00 2001
From: pexcn <pexcn97@gmail.com>
Date: Fri, 14 Feb 2025 15:09:53 +0800
Subject: [PATCH 3/3] add cortex-a53 optimize flags

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cead23e6f..e969ccc75 100644
--- a/Makefile
+++ b/Makefile
@@ -750,7 +750,7 @@ ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS   += -Os
 else
 ifeq ($(cc-name),clang)
-KBUILD_CFLAGS   += -O3
+KBUILD_CFLAGS   += -O3 -finline-hint-functions -march=armv8-a -mtune=cortex-a53 -mcpu=cortex-a53
 else
 KBUILD_CFLAGS   += -O2
 endif
-- 
2.45.2

